### DISCUSSION TITLE
Why is the `our_implementation_used_by_git` test slow on Windows?

### DISCUSSION THREAD

**Discussion by EliahKagan** [GitoxideLabs/gitoxide#1579] [Q&A]
The [`our_implementation_used_by_git`](https://github.com/Byron/gitoxide/blob/d69c6175574f34d6df92b4488ed2c9a85df12c89/gix-filter/tests/driver/mod.rs#L7-L17) test tends to run slowly on Windows. For example, in its [most recent CI run](https://github.com/Byron/gitoxide/actions/runs/10727936688/job/29751377139#step:8:3002), it took almost 5 minutes.

When I run the tests locally on my machine, the situation is sometimes more severe. Recently, it took over 13 minutes:

```text
        PASS [ 783.163s] gix-filter::filter driver::baseline::our_implementation_used_by_git
```

Although I know some file operations are slower on Windows, and some operations that may be performed in shell scripts are slower on Windows if they use a Cygwin/MSYS-like emulation layer (such as if they use the emulated `stat` system call), I'm not sure if that has anything to do with this.

The slowness might be considered a bug, but without more knowledge about why it is happening or what kind of running times are expected, I do not assume that.

**Comment by Byron** [comment]
The only reason I have witnessed for windows tests to be considerable slower was the creation of huge test-cases by means of shell-scripts.
But that wouldn't explain why something is slow in repeated runs.

My assumption is that it's not just a Rust-process spinning hot on CPU or doing a lot of IO, as this would be easily recognisable, but then again, it would have to be in order to explain the issue with repeated runs remains. Or maybe it's writable fixtures that take a long time to recreate? These are certainly harder to see in the task manager, but at least I'd expect a `bash.exe` to be somewhere in the call-tree.

**Reply by EliahKagan** [reply]
Oddly, I didn't see anything taking up a lot of CPU or RAM when it was happening.

Since the cause is not yet known, I will try to figure something out about it.
